app/controllers/reservations_controller.rb

```diff
@@ -21,7 +21,17 @@ def show
   end
 
   def edit
-    render layout: false if request.xhr?
+    if @reservation.requester == current_user
+      if @reservation.modifiable?
+        render layout: false if request.xhr?
+      else
+        flash[:alert] = t('reservations.not_modified')
+        redirect_to reservation_path(@reservation)
+      end
+    else
+      flash[:alert] = t('users.unauthorized')
+      redirect_to root_path
+    end
   end
 
   def update
```

Comment:
Check the conditions in update function also. Does this work: Change the id of the reservation in the edit form and try to submit.